bin/args.js

```diff
@@ -0,0 +1,33 @@
+const yargs = require('yargs');
+
+// Validate and normalize.
+const validate = parser => {
+  const { argv } = parser;
+  const { mdx } = argv;
+
+  return Promise.resolve({ mdx });
+};
+
+const args = () =>
+  yargs
+    .usage(`Usage: spectacle -m <file>`)
+
+    // MDX File
+    .option('mdx', {
```

Comment:
This looks funky to me - `Usage: spectacle -m <file>`. As a spectacle consumer, I would assume that `-m` in that scenario means "file name" but here it's actually `mdx`. A better structure would be to have a filename option like `-s/src` which takes a file of either `.js(x)` or `.md(x)`. And then either infer the filetype from webpack or have a boolean flag `-m/mdx` which sets the filetype to `md(x)`.

It's also weird that the default filename is hidden behind a flag. I would assume you could just run `spectacle` and have smart defaults like "look for slides.js and spin up webserver"

Reply:
What would you expect `spectacle -s <jsx file name here>` to do? AFAIK that's undefined behavior.

Reply:
Run a slideshow from JSX slides. It's currently undefined yes. I think the main point is that the slide file is separate from "is this mdx". And that decision can happen from a user supplied flag, OR we might be able to infer it based on the source file type.

Reply:
Makes sense to me. Here's the decision tree I've got in mind:

Did the user supply a file?

If no, default to (?) `slides.mdx`.

If yes, what kind of file did they supply an mdx file or a jsx file?

If mdx, boot up the presentation.

If jsx (or other), console log about currently unsupported behavior.

Reply:
Sounds reasonable, though I'd eventually expect the default to be `slides.jsx`